source/app/libpd.c

```diff
@@ -329,12 +329,12 @@ char *get_global_cloud_status()
 {
 	char *temp = NULL;
 	int  rv;
-    struct timeval ts;
+    struct timespec ts;
 	pthread_mutex_lock (&cloud_mut);
 	WalPrint("mutex lock in consumer thread\n");
 	WalPrint("Before pthread cond wait in consumer thread\n");
 
-	gettimeofday(&ts, NULL);
+    clock_gettime(CLOCK_MONOTONIC_RAW, &ts);
```

Comment:
It seems timestamp calculation is not correct in this change.
Test logs shows timeout error even though 5min timeout is not reached as per log timestamp.


Test logs -

`250807-09:39:29.711409 [mod=WEBPA, lvl=INFO] [tid=12539] WEBPA: Sent retrieve request successfully to parodus
250807-09:39:29.712384 [mod=WEBPA, lvl=ERROR] [tid=12539] WEBPA: Timeout Error. Unable to get cloud_status even after 5 minutes
250807-09:39:29.712445 [mod=WEBPA, lvl=ERROR] [tid=12539] WEBPA: Received cloud_status as (null), Retrying after backoffRetryTime 3 seconds
250807-09:39:29.712798 [mod=WEBPA, lvl=INFO] [tid=8385] WEBPA: cloud-status Retrieve response received from parodus : {"cloud-status":"online"} len 25 transaction_uuid 0d684f0d-4f5a-44f3-8418-61886f0ce2f2
`